app/models/semester.rb

```diff
@@ -1,16 +1,34 @@
+# frozen_string_literal: true
+
 class Semester < ActiveRecord::Base
   include Authority::Abilities
   has_many :freeform_shows, dependent: :destroy
   has_many :specialty_shows, dependent: :destroy
   has_many :talk_shows, dependent: :destroy
 
+  before_validation :ensure_beginning_and_ending_are_at_six_am
```

Comment:
This does not need to come before the validates calls to run before them, and [conventional style](https://github.com/rubocop-hq/rails-style-guide#macro-style-methods) is to keep callbacks collocated, after validation calls